python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -392,38 +388,21 @@ def fit(self, X, y, convert_dtype=False):
 
         """
 
-        c = default_client()
-
         self.num_classes = len(y.unique())
-        X_futures = workers_to_parts(c.sync(_extract_partitions, X))
-        y_futures = workers_to_parts(c.sync(_extract_partitions, y))
-
-        X_partition_workers = [w for w, xc in X_futures.items()]
-        y_partition_workers = [w for w, xc in y_futures.items()]
-
-        if set(X_partition_workers) != set(self.workers) or \
-           set(y_partition_workers) != set(self.workers):
-            raise ValueError("""
-              X is not partitioned on the same workers expected by RF\n
-              X workers: %s\n
-              y workers: %s\n
-              RF workers: %s
-            """ % (str(X_partition_workers),
-                   str(y_partition_workers),
-                   str(self.workers)))
+
+        data = DistributedDataHandler.create((X, y), client=self.client)
+        self.datatype = data.datatype
 
         futures = list()
-        for w, xc in X_futures.items():
+        for idx, wf in enumerate(data.worker_to_parts.items()):
             futures.append(
-                c.submit(
+                self.client.submit(
                     RandomForestClassifier._fit,
-                    self.rfs[w],
-                    xc,
-                    y_futures[w],
+                    self.rfs[wf[0]],
+                    wf[1],
```

Comment:
remembering what wf[0] and wf[1] mean is always confusing here... can you use destructuring iteration like: 

```
for idx, (worker, worker_data) in enumerate(data.worker_to_parts.items()):
  ...
```
then we have proper names for each item we're submitting